packages/react/src/components.tsx

```diff
@@ -1,15 +1,79 @@
-import type {RemoteComponentType} from '@remote-ui/core';
+import {isValidElement, memo, useMemo, useRef} from 'react';
+import type {ComponentType} from 'react';
+import {isRemoteFragment, RemoteComponentType} from '@remote-ui/core';
 import type {ReactComponentTypeFromRemoteComponentType} from './types';
 
+import {useRender} from './hooks';
+
+interface Options<Props> {
+  fragmentProps?: (keyof Props)[];
+}
+
 export function createRemoteReactComponent<
   Type extends string,
   Props = {},
   AllowedChildren extends RemoteComponentType<string, any> | boolean = true
 >(
   componentType: Type | RemoteComponentType<Type, Props, AllowedChildren>,
+  {fragmentProps}: Options<Props> = {},
 ): RemoteComponentType<Type, Props, AllowedChildren> &
   ReactComponentTypeFromRemoteComponentType<
     RemoteComponentType<Type, Props, AllowedChildren>
   > {
+  if (fragmentProps?.length) {
+    return createComponentWrapper(componentType) as any;
+  }
   return componentType as any;
 }
+
+function createComponentWrapper<T>(componentType: T): T {
+  const Component: ComponentType = componentType as any;
+
+  return memo(function ComponentWrapper({
+    children: externalChildren = [],
+    ...externalProps
+  }: any) {
+    const fragments = useRef<any>({});
```

Comment:
We could probably have a slightly better type here

Reply:
Fixed